### PR TITLE
PoX-4 tests

### PR DESCRIPTION
**Pull request by zone117x**
* Update the stacks-node docker image to latest Nakamato code which boots into epoch 2.5 with `pox-4` enabled: https://github.com/hirosystems/stacks-regtest-env/pull/26
* Update all pox-3 tests to now test pox-4

### CODE REVIEW

File: src/tests-2.5/pox-4-delegate-revoked-stacking.ts [hunk 1]
```diff
@@ -340,14 +351,15 @@ describe('PoX-3 - Delegate Revoked Stacking', () => {
...
+      fee: 10000n,
     });
     const delegateTxResult = await testEnv.client.sendTransaction(
       Buffer.from(delegateTx.serialize())
     );
     const delegateDbTx = await standByForTx(delegateTxResult.txId);
-    expect(delegateDbTx.status).not.toBe(DbTxStatus.Success);
     const delegateResult = decodeClarityValue(delegateDbTx.raw_result);
     expect(delegateResult.repr).toEqual('(err 3)'); // ERR_STACKING_ALREADY_STACKED
+    expect(delegateDbTx.status).not.toBe(DbTxStatus.Success);
```
Comment (zone117x) [inline src/tests-2.5/pox-4-delegate-revoked-stacking.ts:362]: @janniks this test is showing that revoke delegation behavior in pox-4 is different than pox-3. Does this seem like a pox-4 regression to you?

Comment (zone117x) [reply src/tests-2.5/pox-4-delegate-revoked-stacking.ts:362]: It seems to be only this individual test `Try to perform delegate-stx - again`, if skipped then all the rest of the revoke tests pass.

Comment (janniks) [reply src/tests-2.5/pox-4-delegate-revoked-stacking.ts:362]: If I understand correctly, this is intended in pox-4. `Try to perform delegate-stx - again` tests that if still locked a `delegate-stx` will fail.

In pox-4 we can delegate-stx, even if stacking:
https://github.com/stacks-network/stacks-core/blob/de8129cced156bb180bd19e42ac4ad1b2bca8d53/stackslib/src/chainstate/stacks/boot/pox-4.clar#L660-L661

Comment (zone117x) [reply src/tests-2.5/pox-4-delegate-revoked-stacking.ts]: Good catch! I removed that test. All should be passing now

**Review by rafaelcr — APPROVED** [review]
LGTM

File: src/tests-2.5/pox-4-delegate-revoked-stacking.ts [hunk 2]
```diff
@@ -271,9 +276,14 @@ describe('PoX-3 - Delegate Revoked Stacking', () => {
...
       network: testEnv.stacksNetwork,
       anchorMode: AnchorMode.OnChainOnly,
+      fee: 10000n,
     });
     const revokeTxResult = await testEnv.client.sendTransaction(Buffer.from(revokeTx.serialize()));
-    await standByForTxSuccess(revokeTxResult.txId);
+    const revokeStackDbTx = await standByForTxSuccess(revokeTxResult.txId);
+
+    const revokeStackResult = decodeClarityValue(revokeStackDbTx.raw_result);
+    expect(revokeStackResult.repr).toEqual('(ok true)'); // ERR_STACKING_PERMISSION_DENIED
```
Comment (janniks) [inline src/tests-2.5/pox-4-delegate-revoked-stacking.ts:285]: nit: misleading comment?

**Review by janniks — APPROVED** [review]
👏4️⃣

**Pull request merged**